dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateMemoryStore.cs

```diff
@@ -0,0 +1,516 @@
+        (HttpResponseMessage _, string responseContent) = await this.ExecuteHttpRequestAsync(request, cancellationToken).ConfigureAwait(false);
+
+        GetSchemaResponse? getSchemaResponse = JsonSerializer.Deserialize<GetSchemaResponse>(responseContent, s_jsonSerializerOptions);
+
+        foreach (GetClassResponse? @class in getSchemaResponse!.Classes!
+                     .Where(@class => IsPossibleSemanticKernelCollection(@class.Class!)))
+        {
+            yield return @class.Class!;
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task DeleteCollectionAsync(string collectionName, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNullOrEmpty(collectionName, "Collection name is empty");
+
+        if (await this.DoesCollectionExistAsync(collectionName, cancellationToken).ConfigureAwait(false))
+        {
+            string className = ToWeaviateFriendlyClassName(collectionName);
+            this._log.LogTrace("Deleting collection: {0}, with class name: {1}", collectionName, className);
+
+            using HttpRequestMessage request = DeleteSchemaRequest.Create(className).Build();
+            (HttpResponseMessage response, string responseContent) = await this.ExecuteHttpRequestAsync(request, cancellationToken).ConfigureAwait(false);
+
+            try
+            {
+                response.EnsureSuccessStatusCode();
+            }
+            catch (HttpRequestException e)
+            {
+                this._log.LogError(e, "Collection deletion failed: {0}, {1}", e.Message, responseContent);
+                throw;
+            }
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<string> UpsertAsync(string collectionName, MemoryRecord record, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNullOrEmpty(collectionName, "Collection name is empty");
+
+        return await this.UpsertBatchAsync(collectionName, new[] { record }, cancellationToken).FirstOrDefaultAsync(cancellationToken).ConfigureAwait(false) ?? string.Empty;
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<string> UpsertBatchAsync(string collectionName, IEnumerable<MemoryRecord> records,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        Verify.NotNullOrEmpty(collectionName, "Collection name is empty");
+
+        this._log.LogTrace("Upserting vectors");
+
+        string className = ToWeaviateFriendlyClassName(collectionName);
+        BatchRequest requestBuilder = BatchRequest.Create(className);
+        foreach (MemoryRecord? record in records)
+        {
+            requestBuilder.Add(record);
+        }
+
+        using HttpRequestMessage request = requestBuilder.Build();
+        (HttpResponseMessage response, string responseContent) = await this.ExecuteHttpRequestAsync(request, cancellationToken).ConfigureAwait(false);
+
+        response.EnsureSuccessStatusCode();
```

Comment:
Should we add try/catch and re-throw `WeaviateMemoryException` here?

Reply:
Done